tests/util/localregistry/cmd/localregistry.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/mixer/cmd/shared"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	// Default values for test env setup
+	LocalRegistryFile      = "tests/util/localregistry/localregistry.yaml"
+	LocalRegistryNamespace = "kube-system"
+	LocalRegistryPort      = 5000
+)
+
+type localRegistrySetupArgs struct {
+	remoteRegistryPort uint16
+	kubeconfig         string
+}
+
+func setupCfgCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
+	sa := &localRegistrySetupArgs{}
+	setupCmd := &cobra.Command{
+		Use:   "setup",
+		Short: "sets up localregistry in kubernetes cluster",
+		Run: func(cmd *cobra.Command, args []string) {
+			SetupLocalRegistry(sa.remoteRegistryPort, sa.kubeconfig, printf, fatalf)
+		},
+	}
+	setupCmd.PersistentFlags().Uint16Var(&sa.remoteRegistryPort, "remoteRegistryPort", 5000,
+		"port number where registry can be port-forwarded on user's machine. Default is 5000, same as local "+
+			"registry port.")
+	setupCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
+		"Use a Kubernetes configuration file instead of in-cluster configuration")
+	return setupCmd
+}
+
+func teardownCfgCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
+	sa := &localRegistrySetupArgs{}
+	teardownCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "deletes localregistry in kubernetes cluster",
+		Run: func(cmd *cobra.Command, args []string) {
+			TeardownLocalRegistry(sa.kubeconfig, printf, fatalf)
+		},
+	}
+	teardownCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
+		"Use a Kubernetes configuration file instead of in-cluster configuration")
+	return teardownCmd
+}
+
+// GetLocalRegistry sets up localregistry and return th registry
+func SetupLocalRegistry(remoteRegistryPort uint16, kubeconfig string, printf, fatalf shared.FormatFn) {
```

Comment:
perhaps this method should return an error, rather than relying on `fatalf` to terminate the process?